lib/tests/2wp.js

```diff
@@ -175,13 +175,82 @@ const execute = (description, getRskHost) => {
 
     });
 
+    it('should do legacy pegin with multiple inputs from different accounts and two outputs to the federation with value exactly minimum', async () => {
+
+      // Arrange
+
+      const initial2wpBalances = await get2wpBalances(rskTxHelper, btcTxHelper);
+      const senderRecipientInfo1 = await createSenderRecipientInfo(rskTxHelper, btcTxHelper);
+      const senderRecipientInfo2 = await createSenderRecipientInfo(rskTxHelper, btcTxHelper);
+      const initialSender1AddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo1.btcSenderAddressInfo.address);
+      const initialSender2AddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo2.btcSenderAddressInfo.address);
+
+      const sender1PeginValueInSatoshis = minimumPeginValueInSatoshis;
+      const sender2PeginValueInSatoshis = minimumPeginValueInSatoshis;
+      const peginValueInSatoshis = sender1PeginValueInSatoshis + sender2PeginValueInSatoshis;
+
+      const sender1UtxosInfo = await getAddressUtxosInfo(btcTxHelper, senderRecipientInfo1.btcSenderAddressInfo.address, satoshisToBtc(sender1PeginValueInSatoshis));
+      const sender2UtxosInfo = await getAddressUtxosInfo(btcTxHelper, senderRecipientInfo2.btcSenderAddressInfo.address, satoshisToBtc(sender2PeginValueInSatoshis));
+
+      const sender1ChangeInSatoshis = btcToSatoshis(sender1UtxosInfo.change);
+      const sender2ChangeInSatoshis = btcToSatoshis(sender2UtxosInfo.change);
+
+      const tx = new bitcoinJsLib.Transaction();
+
+      // Adding inputs
+      addInputs(tx, sender1UtxosInfo);
+      addInputs(tx, sender2UtxosInfo);
+
+      // Adding 2 outputs to the federation
+      addOutputToFed(tx, sender1PeginValueInSatoshis);
+      addOutputToFed(tx, sender2PeginValueInSatoshis);
+
+      // Adding change outputs
+      addChangeOutput(tx, senderRecipientInfo1.btcSenderAddressInfo.address, sender1ChangeInSatoshis);
+      addChangeOutput(tx, senderRecipientInfo2.btcSenderAddressInfo.address, sender2ChangeInSatoshis);
+
+      const sendersPrivateKeys = [senderRecipientInfo1.btcSenderAddressInfo.privateKey, senderRecipientInfo2.btcSenderAddressInfo.privateKey];
+
+      const signedTx = await btcTxHelper.nodeClient.signTransaction(tx.toHex(), [], sendersPrivateKeys);
+
+      // Act
+
+      const btcPeginTxHash = await btcTxHelper.nodeClient.sendTransaction(signedTx);
+      // Assert
+
+      // Since we are not using `sendPegin` here, we need to do some extra steps before ensuring the pegin is registered.
+      const expectedCountOfThisPeginUtxosInTheBridge = 2;
```

Comment:
If we are expected 2 outputs to the federation in this pegin, we need to explicitly let the `ensurePeginIsPushed` know about it. Otherwise, it will fail if it finds a different number of utxos with this pegin btc hash.